src/operun/contactform/browser/contactform.py

```diff
@@ -0,0 +1,179 @@
+            # Else, field has no value, so check if it is required instead
+            elif field in self.required:
+                self.errors.append(field)
+        if self.errors:
+            self.show_validate_error_message()
+            return False
+        return True
+
+    def _send_mail(self):
+        """
+        Send E-Mails to admin and user.
+        """
+        form_data = dict()
+        # Create a dictionary of form values to be
+        # passed to the E-Mail templates.
+        # We DO NOT pass "self.form" directly, as it could contain unwanted
+        # fields or submitted data.
+        for field in self.fields:
+            form_data[field] = self.form.get(field, '')
+        # Assign some default values
+        portal_address = api.portal.get_registry_record(
+            'plone.email_from_address'
+        )
+        form_address = self.form.get('email', '')
+        form_subject = self.form.get('subject', '')
+        # We create a list of recipients with their associated E-Mail
+        # template, and "send_email" configuration.
+        mailing_list = {
+            'admin': {
+                'template': '@@mailto_admin',
+                'subject': form_subject,
+                'sender': form_address,
+                'recipient': portal_address,
+            },
+            'user': {
+                'template': '@@mailto_user',
+                'subject': u'operun contact confirmation',
+                'sender': portal_address,
+                'recipient': form_address,
+            },
+        }
+        for user in mailing_list:
+            try:
+                # We fetch the registered E-Mail template...
+                mail_view = self.context.restrictedTraverse(
+                    mailing_list[user]['template']
+                )
+                # We parse it with our "form_data" values...
+                mail_body = mail_view(
+                    self.context,
+                    header=mailing_list[user]['subject'],
+                    **form_data
+                ).encode('utf-8')
+                # We use MIMEMultipart to create an HTML
+                # Content-Disposition header.
+                # See: https://docs.plone.org/develop/plone.api/docs/portal.html#send-e-mail  # noqa: 501
+                message = MIMEMultipart('alternative')
+                part = MIMEText(mail_body, 'html', 'utf-8')
+                message.attach(part)
+                # We attempt to send the E-Mail, note we don't use "immediate"
+                # as this can fail in a production environment.
+                api.portal.send_email(
+                    body=message.as_string(),
+                    subject=mailing_list[user]['subject'],
+                    sender=mailing_list[user]['sender'],
+                    recipient=mailing_list[user]['recipient'],
+                )
+            except Exception as err:
+                logger.error(err)
```

Comment:
Not sure whether there is an issue here, but you should check that the `err` is really raised. I had an issue in actionbike, that the successpage was shown, even when the emails were not sent. This was because the exception was only raised on transaction commit or when `immediate=True`... 
To check this: Just put `localhost` as a mail server in the control_panel and try sending emails with your code.